tests/s25Main/UI/testSmartBitmap.cpp

```diff
@@ -26,6 +30,77 @@ static std::ostream& boost_test_print_type(std::ostream& os, const ColorBGRA col
 }
 } // namespace libsiedler2
 
+namespace rttrOglMock2 {
+RTTR_IGNORE_DIAGNOSTIC("-Wmissing-declarations")
+
+const Point<float>* texturePointer;
+const Point<float>* vertexPointer;
+std::vector<Point<float>> textureCoords;
+std::vector<Point<float>> vertexCoords;
+int callCount;
+
+void APIENTRY glVertexPointer(GLint size, GLenum type, GLsizei stride, const void* pointer)
+{
+    ++callCount;
+    RTTR_Assert(type == GL_FLOAT);
+    RTTR_Assert(stride == 0);
+    RTTR_Assert(size == 2);
+    vertexPointer = static_cast<const Point<float>*>(pointer);
+}
+
+void APIENTRY glDrawArrays(GLenum mode, GLint first, GLsizei count)
+{
+    ++callCount;
+    RTTR_Assert(mode == GL_QUADS);
+    RTTR_Assert(texturePointer != nullptr);
+
+    textureCoords.resize(count);
+    std::copy(&texturePointer[first], &texturePointer[first + count], textureCoords.begin());
+
+    vertexCoords.resize(count);
+    std::copy(&vertexPointer[first], &vertexPointer[first + count], vertexCoords.begin());
+}
+
+void APIENTRY glEnableClientState(GLenum array)
+{
+    RTTR_UNUSED(array);
+    ++callCount;
+}
+
+void APIENTRY glTexCoordPointer(GLint size, GLenum type, GLsizei stride, const void* pointer)
+{
+    ++callCount;
+    RTTR_Assert(type == GL_FLOAT);
+    RTTR_Assert(stride == 0);
+    RTTR_Assert(size == 2);
+    texturePointer = static_cast<const Point<float>*>(pointer);
+}
+
+void APIENTRY glColorPointer(GLint size, GLenum type, GLsizei stride, const void* pointer)
+{
+    ++callCount;
+    RTTR_UNUSED(size);
```

Comment:
Just FYI: You can just remove the parameter names if you only care about the callcount to avoid the unused macros. Unless you want them to be shown, but if you don't use any I don't see a reason for the names